narwhal/primary/src/aggregators.rs

```diff
@@ -57,11 +64,36 @@ impl VotesAggregator {
 
         if self.weight >= committee.quorum_threshold() {
             self.weight = 0; // Ensures quorum is only reached once.
-            return Ok(Some(Certificate::new(
-                committee,
-                header.clone(),
-                self.votes.clone(),
-            )?));
+            let cert = Certificate::new(committee, header.clone(), self.votes.clone())?;
+            let (_, pks) = cert.signed_by(committee);
+
+            let certificate_digest: Digest<{ crypto::DIGEST_LENGTH }> = Digest::from(cert.digest());
+            match cert
+                .aggregated_signature
+                .verify_secure(&to_intent_message(certificate_digest), &pks[..])
```

Comment:
i cannot directly uses https://github.com/MystenLabs/sui/blob/776c90b88c87d86102bd31b41c320025c2d0de39/narwhal/types/src/primary.rs#L640 bc there is no worker_cache here - is that something we need to verify?

Reply:
Verifying information outside of votes should be unnecessary here, since Header is produced locally.